### PR TITLE
[3.12] Add zizmor to pre-commit and fix most findings (GH-127749)

### PR DESCRIPTION
**Pull request by hugovk**
(cherry picked from commit ae31df354d02e12bf656954c5c72380d96c1dc0e)

### CODE REVIEW

File: .github/workflows/reusable-windows.yml [hunk 1]
```diff
@@ -42,6 +46,6 @@ jobs:
       if: inputs.arch != 'arm64'
       run: >-
         .\PCbuild\rt.bat
-        -p ${{ inputs.arch }}
+        -p "${ARCH}"
         -d -q -uall -u-cpu -rwW
         --slowest --timeout=1200 -j0
```
Comment (ezio-melotti) [inline .github/workflows/reusable-windows.yml]:
```suggestion
        --slowest --timeout=1200 -j0
      shell: bash
```

Comment (ezio-melotti) [reply .github/workflows/reusable-windows.yml]: I went ahead and committed this to see if it fixes the problem.

File: .github/workflows/reusable-windows.yml [hunk 2]
```diff
@@ -42,6 +46,7 @@ jobs:
       if: inputs.arch != 'arm64'
       run: >-
         .\PCbuild\rt.bat
```
Comment (ezio-melotti) [inline .github/workflows/reusable-windows.yml]:
```suggestion
        .\\PCbuild\\rt.bat
```

File: .github/workflows/reusable-windows.yml [hunk 3]
```diff
@@ -24,24 +24,29 @@ jobs:
...
     steps:
     - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
     - name: Register MSVC problem matcher
       if: inputs.arch != 'Win32'
       run: echo "::add-matcher::.github/problem-matchers/msvc.json"
     - name: Build CPython
       run: >-
         .\PCbuild\build.bat
```
Comment (ezio-melotti) [inline .github/workflows/reusable-windows.yml]:
```suggestion
        .\\PCbuild\\build.bat
```

File: .github/workflows/reusable-windows.yml [hunk 4]
```diff
@@ -24,24 +24,29 @@ jobs:
...
         -e -d
-        -p ${{ inputs.arch }}
+        -p "${ARCH}"
         ${{ fromJSON(inputs.free-threading) && '--disable-gil' || '' }}
```
Comment (ezio-melotti) [inline .github/workflows/reusable-windows.yml]:
```suggestion
        ${{ fromJSON(inputs.free-threading) && '--disable-gil' || '' }}
      shell: bash
```

**Review by ezio-melotti — APPROVED** [review]

**Comment by hugovk** [comment]
Thanks @ezio-melotti for fixing this up :)

**Comment by hugovk** [comment]
GitHub won't let me merge this right now: "Merge already in progress".

Let's try later...

File: .github/workflows/reusable-windows-msi.yml [hunk 5]
```diff
@@ -17,8 +17,11 @@ jobs:
...
     env:
+      ARCH: ${{ inputs.arch }}
       IncludeFreethreaded: true
     steps:
     - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
     - name: Build CPython installer
-      run: .\Tools\msi\build.bat --doc -${{ inputs.arch }}
+      run: .\Tools\msi\build.bat --doc -"${ARCH}"
```
Comment (AlexWaygood) [inline .github/workflows/reusable-windows-msi.yml]:
```suggestion
      run: .\Tools\msi\build.bat --doc -"${ARCH}"
      shell: bash
```

**Review by AlexWaygood — COMMENTED** [review]
One more Windows workflow where I think you need to specify `shell: bash`. Not sure why it's _not_ failing on this PR but I think we should specify bash anyway:

File: .github/workflows/reusable-windows-msi.yml [hunk 6]
```diff
@@ -25,3 +25,4 @@ jobs:
         persist-credentials: false
     - name: Build CPython installer
       run: .\Tools\msi\build.bat --doc -"${ARCH}"
```
Comment (AlexWaygood) [inline .github/workflows/reusable-windows-msi.yml]: I... guess this isn't valid bash now? Do we have to do this?

```suggestion
      run: ./Tools/msi/build.bat --doc -"${ARCH}"
```

**Pull request merged**